src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -0,0 +1,188 @@
+        try:
+            output = process.run(**kwargs)
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(lstopo, error)
+            self.fail(msg)
+
+        # Separate numa node info, pop first item as it has no useful info
+        numas = re.split("NUMANode", output.stdout)
+        numas.pop(0)
+
+        numas_devs = {}
+        for i, numa in enumerate(numas):
+            net_dev = []
+            split_numa = numa.split("\n")
+            for line in split_numa:
+                if "Net" in line or "OpenFabrics" in line:
+                    # Get the devices i.e. eth0, ib0, ib1
+                    net_dev.append("".join(line.strip().split('"')[1::2]))
+            numas_devs[i] = net_dev
+
+        return numas_devs
+
+    def get_dev_provider(self, device):
+        """Get expected values of provider with fi_info."""
+        fi_info = os.path.join(
+            self.bin, "fi_info -d {} | grep provider | sort -u".format(device))
+        kwargs = {
+            "cmd": fi_info,
+            "allow_output_check": "combined",
+            "verbose": False,
+            "shell": True,
+        }
+        try:
+            output = process.run(**kwargs)
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(fi_info, error)
+            self.fail(msg)
+
+        # Clean up string
+        provider = "none"
+        if output.stdout != "":
+            provider = output.stdout.splitlines()[0].split()[-1]
+
+        return provider
+
+    def test_dmg_network_scan_basic(self):
+        """
+        JIRA ID: DAOS-2516
+        Test Description: Test basic dmg functionality to scan the network
+        devices on the system.
+        :avocado: tags=all,tiny,pr,hw,dmg,network_scan,basic
+        """
+        # Create dmg command
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.get_params(self)
+
+        # Update hostlist value for dmg command
+        port = self.params.get("port", "/run/server_config/*")
+        servers_with_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+
+        try:
+            dmg_out = dmg.run()
+        except process.CmdError as details:
+            self.fail("dmg command failed: {}".format(details))
+
+        # This parse asumes that the device information is listed on separate
+        # lines and with this format 'info_type: info'
+        if isinstance(dmg_out.stdout, str):
+            pout = {}
+            net_items = ["numa_node:", "iface:", "provider:"]
+            for i, line in enumerate(dmg_out.stdout.splitlines()):
+                for j, item in enumerate(net_items):
+                    if item in line:
+                        pout[i + j] = line.strip('\t')
```

Comment:
Seeing as this is a test for the `dmg network scan` command it is probably not ideal to assume that the output from the command is correct.  If for some reason the device information did not appear on separate lines or was inconsistent in order or data (not all entries have a "numa_node" entry for example) you could get data corruption in your `pout` dictionary.   At the very least the code should probably fail if it is about to overwrite an existing `pout` key.  You could also verify the output format with regular expression command line matching and use the command output to define your keys - with the same key-overwrite-checking of course.

Reply:
The reason I made the key to be [i + j] was to make sure I would have separate keys. Since i represents the line number and j represents the net_item it finds in the line. The output usually looks like this:
```
[
        fabric_iface: ib0
        provider: ofi+psm2
        pinned_numa_node: 0

        fabric_iface: ib0
        provider: ofi+sockets
        pinned_numa_node: 0

        fabric_iface: eth0
        provider: ofi+sockets
        pinned_numa_node: 0

        fabric_iface: lo
        provider: ofi+sockets
        pinned_numa_node: 0
]
```
Since 'i' will correspond to a different line number as it iterates through the lines, making the keys unique. Also, the reason I used a dict to store the output of the dmg command was to differentiate, with the keys, the different items regardless of whether the content of the value was the same.

One thing that I should check for is that if I can't find any of the net_items in the lines, the test should fail.